src/Components/Web/src/Routing/NavLink.cs

```diff
@@ -17,41 +17,41 @@ public class NavLink : ComponentBase, IDisposable
         private const string DefaultActiveClass = "active";
 
         private bool _isActive;
-        private string _hrefAbsolute;
-        private string _class;
+        private string? _hrefAbsolute;
+        private string? _class;
 
         /// <summary>
         /// Gets or sets the CSS class name applied to the NavLink when the
         /// current route matches the NavLink href.
         /// </summary>
         [Parameter]
-        public string ActiveClass { get; set; }
+        public string? ActiveClass { get; set; }
 
         /// <summary>
         /// Gets or sets a collection of additional attributes that will be added to the generated
         /// <c>a</c> element.
         /// </summary>
         [Parameter(CaptureUnmatchedValues = true)]
-        public IReadOnlyDictionary<string, object> AdditionalAttributes { get; set; }
+        public IReadOnlyDictionary<string, object?>? AdditionalAttributes { get; set; }
```

Comment:
Do we use this annotation on all AdditionalAttributes parameters? I think I've seen one that is different, but I can't find it now. Just double check that these are consistent across different components.

Does the current annotation mean that the dictionary and its contents can be both null? (So you have a valid entry with a null value) What's the scenario for that last case?

Reply:
> Do we use this annotation on all AdditionalAttributes parameters?

Yup, the compiler should complain if we missed some.

> Does the current annotation mean that the dictionary and its contents can be both null? (So you have a valid entry with a null value) What's the scenario for that last case?

I went back and checked, and I don't think there's a scenario for the values inside the dictionary being null. I can change that. The dictionary itself would be null.

Reply:
Ok, I see what happened here. `RenderTreeBuilder` allows null-valued attributes here: https://github.com/dotnet/aspnetcore/blob/master/src/Components/Components/src/Rendering/RenderTreeBuilder.cs#L329-L333. Making https://github.com/dotnet/aspnetcore/blob/master/src/Components/Components/src/Rendering/RenderTreeBuilder.cs#L405 allow nullable attributes prevents these properties from having a non-nullable value. 

It's a bit of a compromise, but we'll declare AddMultipleAttributes to accept non-null values. `ComponentProperties` already produces a non-null valued dictionary, so we should be good.